src/scanner/scanner.py

```diff
@@ -0,0 +1,319 @@
+                "Options: csv, parquet or duckdb"
+            )
+
+        output_dir = Path(output_dir)
+        output_dir.mkdir(parents=True, exist_ok=True)
+
+        file_name = self.uf + "_" + self.disease + "." + format
+        file = output_dir / file_name
+
+        df = self._parse_results()
+
+        if file.exists() and format != "duckdb":
+            logger.warning(f"Overriding {file}")
+            file.unlink()
+
+        try:
+            match format:
+                case "csv":
+                    df.to_csv(file)
+                case "parquet":
+                    df.to_parquet(file)
+                case "duckdb":
+                    self._to_duckdb(output_dir)
+                    file = output_dir / "episcanner.duckdb"
+
+            logger.info(f"Data exported successfully to {file}")
+        except (FileNotFoundError, PermissionError) as e:
+            raise ValueError(f"Failed to write file: {e}")
+        except Exception as e:
+            raise ValueError(f"Unexpected error while writing file: {e}")
+
+    def _get_alerta_table(self) -> pd.DataFrame:
+        """
+        Pulls the HistoricoAlerta data for a disease and UF from the InfoDengue
+        database.
+
+        Returns
+        -------
+            df: Pandas dataframe
+        """
+
+        if self.disease == "dengue":
+            table_suffix = ""
+        else:
+            table_suffix = "_" + self.disease
+
+        state_name = STATES[self.uf]
+
+        query = f"""
+            SELECT historico.*
+            FROM "Municipio"."Historico_alerta{table_suffix}" historico
+            JOIN "Dengue_global"."Municipio" municipio
+            ON historico.municipio_geocodigo=municipio.geocodigo
+            WHERE municipio.uf=\'{state_name}\'
+            ORDER BY "data_iniSE" DESC;
+        """
+
+        with make_connection().connect() as conn:
+            df = pd.read_sql_query(query, conn, index_col="id")
+
+        df.data_iniSE = pd.to_datetime(df.data_iniSE)
+        df.set_index("data_iniSE", inplace=True, drop=True)
+        return df
+
+    def _filter_city(self, geocode):
+        dfcity = self.data[self.data.municipio_geocodigo == geocode].copy()
+        dfcity.sort_index(inplace=True)
+        dfcity["casos_cum"] = dfcity.casos.cumsum()
+        return dfcity
+
+    def _save_results(self, geocode, year, results, curve):
+        self.results[geocode].append(
+            {
+                "year": year,
+                "success": results.success,
+                "params": results.params.valuesdict(),
+                "sir_pars": get_SIR_pars(results.params.valuesdict()),
+            }
+        )
+        self.curves[geocode].append(
+            {
+                "year": year,
+                "df": curve,
+                "residuals": abs(curve.richards - curve.casos_cum),
+                "sum_res": (
+                    sum(abs(curve.richards - curve.casos_cum))
+                    / max(curve.casos_cum)
```

Comment:
@fccoelho @eduardocorrearaujo this line is returning me `ZeroDivisionError: float division by zero`, what should I do in these cases?

Reply:
In this case, if `max(curve.casos_cum` is zero, there is something wrong with either the code, or the fit of the model. @eduardocorrearaujo, could you check the cases where this is happening?

Reply:
I can't provide the data from postgres I'm using because I'm using a local postgres db, but the traceback is:

```py
In [1]: from src.scanner.scanner import EpiScanner

In [2]: x = EpiScanner("dengue", "AL")

In [3]: x.export("duckdb")
2023-12-18 16:41:25.242 | INFO     | src.scanner.scanner:export:140 - Data exported successfully to /home/bida/episcanner/episcanner.duckdb

In [4]: x = EpiScanner("dengue", "AM")
---------------------------------------------------------------------------
ZeroDivisionError                         Traceback (most recent call last)
Cell In[4], line 1
----> 1 x = EpiScanner("dengue", "AM")

File ~/Projetos/InfoDengue/episcanner-downloader/src/scanner/scanner.py:91, in EpiScanner.__init__(self, disease, uf, verbose)
     88 self.data = self._get_alerta_table()
     89 self.window = int(self.data.SE.max() % 100)
---> 91 asyncio.run(self._scan_all())

File ~/micromamba/envs/episcanner-downloader/lib/python3.10/asyncio/runners.py:44, in run(main, debug)
     42     if debug is not None:
     43         loop.set_debug(debug)
---> 44     return loop.run_until_complete(main)
     45 finally:
     46     try:

File ~/micromamba/envs/episcanner-downloader/lib/python3.10/asyncio/base_events.py:649, in BaseEventLoop.run_until_complete(self, future)
    646 if not future.done():
    647     raise RuntimeError('Event loop stopped before Future completed.')
--> 649 return future.result()

File ~/Projetos/InfoDengue/episcanner-downloader/src/scanner/scanner.py:301, in EpiScanner._scan_all(self)
    296 async def _scan_all(self):
    297     tasks = [
    298         self._scan(geocode)
    299         for geocode in self.data.municipio_geocodigo.unique()
    300     ]
--> 301     await asyncio.gather(*tasks)

File ~/Projetos/InfoDengue/episcanner-downloader/src/scanner/scanner.py:294, in EpiScanner._scan(self, geocode)
    285             logger.info(
    286                 f"""
    287                     R0 in {y}: {
   (...)
    290                 """
    291             )
    293 tasks = [scan_year(y) for y in set(df.year.values)]
--> 294 await asyncio.gather(*tasks)

File ~/Projetos/InfoDengue/episcanner-downloader/src/scanner/scanner.py:281, in EpiScanner._scan.<locals>.scan_year(y)
    273     return
    275 out, curve = otim(
    276     dfy[["casos", "casos_cum"]].iloc[0: self.window],  # NOQA E203
    277     0,
    278     self.window,
    279 )
--> 281 self._save_results(geocode, y, out, curve)
    283 if out.success:
    284     if self.verbose:

File ~/Projetos/InfoDengue/episcanner-downloader/src/scanner/scanner.py:200, in EpiScanner._save_results(self, geocode, year, results, curve)
    185 def _save_results(self, geocode, year, results, curve):
    186     self.results[geocode].append(
    187         {
    188             "year": year,
   (...)
    192         }
    193     )
    194     self.curves[geocode].append(
    195         {
    196             "year": year,
    197             "df": curve,
    198             "residuals": abs(curve.richards - curve.casos_cum),
    199             "sum_res": (
--> 200                 sum(abs(curve.richards - curve.casos_cum))
    201                 / max(curve.casos_cum)
    202             ),
    203             "ep_time": comp_duration(curve),
    204         }
    205     )

ZeroDivisionError: float division by zero
```

Reply:
Thanks luã, but we need to check the curve record to diagnose why we have a zero, where we shouldn't